### DISCUSSION TITLE
git diff --name-only and spaces

### DISCUSSION THREAD

**Discussion by eshlomi1** [community/community#21723] [New to GitHub]
<p>Hi</p>
<p>I have a commit that includes several files</p>
<p>One of them is a text file that i would like to commit as new instead of the previous one.</p>
<p>This file has allot of changes from the previous version - and the changes are not readble.</p>
<p>I would like that, once i use</p>
<p>git diff</p>
<p>or</p>
<p>git format-patch</p>
<p>The diff file created - will delete the previous version of this specific file and add the entire modified file.</p>
<p>As for the rest of the file commited - it will use the diff as it is</p>
<p>is there a way to do it?</p>

**Comment by Wabri** [comment, marked as answer]
<p>Hi <a class="mention" href="/eshlomi1">@eshlomi1</a>,</p>
<p>Welcome to the Github Community Forum!</p>
<p>This is a little triky, but i think we can get on it…</p>
<p>From now on i will call <strong>B</strong> the commit that you have create last and <strong>A</strong> the previous commit, and also consider to start from the <strong>master</strong> branch.</p>
<p>With a <strong>git log</strong> you can see the <strong>id of the commit</strong> , something like this:</p>
<pre><code>commit **cb2fa6226febb3f3b6be53b15bba4c83adeaff9f \&lt;-------- This is the id** Author: User name &lt;usermail@mail.com&gt;
Date: Thu Feb 14 16:04:07 2019 +0100

    update
</code></pre>
<p>Now create a branch from A with the git command</p>
<pre><code>git checkout -b &lt;name_of_branch&gt; &lt;commit_id&gt;
</code></pre>
<p>In my case i will use this:</p>
<pre><code>git checkout -b test.one **cb2fa6226febb3f3b6be53b15bba4c83adeaff9f**
</code></pre>
<p>(You can also use the <strong>HEAD^</strong> for the commit id, this refer to the previous commit of the HEAD commit)</p>
<p>Now i have this new branch called <strong>test.one</strong> and I want to import the file _ <strong>test.txt</strong> _ for example.</p>
<p>The <strong>merge --squash</strong> we are going to use is the merge command but without commit:</p>
<pre><code>git merge --squash **master**
</code></pre>
<p>This will import all the changes made with the commit B.</p>
<p>Now you can take 2 ways:</p>
<ol>
<li>
<p>Use this branch like a substitute of the previous one (if is not the master branch);</p>
</li>
<li>
<p>Reset the branch master to A, do changes in the test.one branch than merge.</p>
</li>
</ol>
<p><strong>Way 1</strong></p>
<p>This works only if the main branch is not the master, if it is go to Way 2 below.</p>
<p>Reset all the files:</p>
<pre><code>git reset **cb2fa6226febb3f3b6be53b15bba4c83adeaff9f**
</code></pre>
<p>And do wherever you want like create a new commit with only the <strong><em>test.txt</em></strong>. Do not delete the other branch before you’ve concluded. When you are sure to commit all the changes you want in this branch you can remove the other branch with this command:</p>
<pre><code>git branch -D &lt;the_other_branch&gt;
</code></pre>
<p>And if you also have the same refs in the remote you need to use this:</p>
<pre><code>git push origin --delete &lt;the_other_branch&gt;
</code></pre>
<p>And you have done.</p>
<p><strong>Way 2</strong></p>
<p>Do your commits in this new branch. When you have done and you want this changes in the other branch you can checkout in the previous branch, here i will use the master:</p>
<pre><code>git checkout master
</code></pre>
<p>Now you need to remove the old commit B and move to the previous one in this example we call that A:</p>
<pre><code>git reset --hard **cb2fa6226febb3f3b6be53b15bba4c83adeaff9f**
</code></pre>
<p>Let’s merge the branch testOne:</p>
<pre><code>git merge test.one
</code></pre>
<p>Now you need to force push this master into the remote version:</p>
<pre><code>git push --force origin master
</code></pre>
<p>(Be carefull to use this command because this update the remote history for you and all of your collaborators)</p>
<p>This is it.</p>
<p>Hope this helps, ask if you have more questions :slight_smile:</p>
<p>-Gabriele-</p>
<p>Mark helpfull posts with Accept as Solution to help other users locate important info. Don’t forget to give Kudos for great contents!</p>
